src/main.rs

```diff
@@ -209,20 +202,102 @@ fn to_ca_certs(ca_certs_pem: &Vec<Vec<u8>>) -> Result<rustls::RootCertStore> {
     Ok(root_cert)
 }
 
-/// Initializes the tracing subscriber.
+/// Manages the log file and guards.
 ///
-/// If `log_dir` is `None` or the runtime is in debug mode, logs will be printed to stdout.
+/// `_guards` will flush the logs when they are dropped.
 ///
-/// Returns a vector of `WorkerGuard` that flushes the log when dropped.
-fn init_tracing(log_dir: Option<&Path>) -> Vec<WorkerGuard> {
-    let mut guards = vec![];
-    let subscriber = tracing_subscriber::Registry::default();
-    let file_name = format!("{}.log", env!("CARGO_PKG_NAME"));
+/// `dynamic_log_file_writer` wraps the log file to allow changing its path dynamically.
```

Comment:
@danbi2990 
I’d like to suggest separating the `tracing`-related code into its own file, as it seems like a good candidate for modularization. The `tracing`-related code includes `LogManager`, `DynamicLogFileWriter`, and `init_tracing`.

- On a simple note, it’s longer than `settings.rs` in terms of line count. 😁
- Given the current logging policy, many projects share the same "domain logic." Separating this into its own file could make it much easier to review and update the "domain logic" without distractions.

P.S. I noticed the status was changed to WIP while writing this comment, but I decided to share it anyway since it might still be useful. 😊

Reply:
Thank you for the suggestion. It will be applied. 👍